src/modules/users/users.repository.ts

```diff
@@ -0,0 +1,40 @@
+import { User } from './interfaces/user';
+import { UserSchema } from '../../schemas/users-schema';
+
+class UserRepository {
+  public async findByEmail(email: User['email']): Promise<User> {
```

Comment:
why we have here Class method and public, but below we have class properties?
choose 1 was of representation and let's follow this convention everywhere